ckanext/geonode/harvesters/client.py

```diff
@@ -35,13 +34,9 @@ def _get_resources(self, resType):
         url = '%s/api/%s/' % (self.baseurl, resType)
 
         log.info('Retrieving %s at GeoNode URL %s', resType, url)
-        request = urllib2.Request(url)
-        opener = urllib2.build_opener(urllib2.HTTPCookieProcessor(), urllib2.HTTPRedirectHandler())
-
-        response = opener.open(request)
-        content = response.read()
+        response = urlopen(url)
```

Comment:
For HTTP and HTTPS URLs, this function returns a http.client.HTTPResponse object slightly modified.

[https://docs.python.org/3/library/urllib.request.html](https://docs.python.org/3/library/urllib.request.html#:~:text=For%20HTTP%20and%20HTTPS%20URLs%2C%20this%20function%20returns%20a%20http.client.HTTPResponse%20object%20slightly%20modified.)

Reply:
Should be replaced by

```
        response = urlopen(url).read()
```